components/application-broker/internal/director/service.go

```diff
@@ -0,0 +1,312 @@
+package director
+
+import (
+	"context"
+	"encoding/json"
+	"time"
+
+	"github.com/kyma-project/kyma/components/application-broker/internal"
+
+	schema "github.com/kyma-incubator/compass/components/director/pkg/graphql"
+	"github.com/kyma-project/kyma/components/application-gateway/pkg/authorization"
+	"github.com/kyma-project/kyma/components/application-gateway/pkg/proxyconfig"
+	"github.com/pkg/errors"
+	"k8s.io/apimachinery/pkg/util/wait"
+)
+
+const contextNameForInstanceID = "instance_id"
+
+// APIPackageInstanceAuthClient defines required Director client functionality to handle the
+// package instance auth creation and deletion
+type APIPackageInstanceAuthClient interface {
+	RequestPackageInstanceAuth(context.Context, RequestPackageInstanceAuthInput) (*RequestPackageInstanceAuthOutput, error)
+	FindPackageInstanceAuths(context.Context, GetPackageInstanceAuthsInput) (*GetPackageInstanceAuthsOutput, error)
```

Comment:
maybe FindPackageInstanceAuthsInput and FindPackageInstanceAuthsOutput, the same for second Find method